src/transport/PairingSession.cpp

```diff
@@ -34,16 +34,25 @@ CHIP_ERROR PairingSession::AllocateSecureSession(SessionManager & sessionManager
 
 CHIP_ERROR PairingSession::ActivateSecureSession(const Transport::PeerAddress & peerAddress)
 {
-    Transport::SecureSession * secureSession = mSecureSessionHolder->AsSecureSession();
+    // TODO(17568): Replace with proper expiry logic. This current method makes sure there
+    // are not multiple sessions established, until eventual exhaustion of the resources
+    // for CASE sessions. Current method is quick fix for #17698, cannot remain.
+    mSessionManager->ExpireAllPairingsForPeerExceptPending(GetPeer());
```

Comment:
This seems fine if it can't accidentally expire PASE sessions....

Reply:
It cannot, since it would only expire another PASE session "with the same peer"  that is still active. For PASE, I guess this enforces a single PASE session (which is already a constraint)